src/middleware/router.coffee

```diff
@@ -166,6 +183,7 @@ buildNonPrimarySendRequestPromise = (ctx, route, options, path) ->
 
     ctx.routes = [] if not ctx.routes
     ctx.routes.push routeObj
```

Comment:
If we make the changes I've mentioned above then we won't need to attach the routes to the context at all, we could just pass the routeObj directly to the relevent messageStore function.